### ISSUE TITLE
Raygun iOS not catching crashes 

### ISSUE THREAD

**Issue by wiljoetradify** [MindscapeHQ/raygun4maui#9]
Hi Mindscape team

I have found that iOS app do not log exception in Raygun

I set up a modified project from your SampleApp. I did put the key into the API Key but nothing get logged into my Raygun bucket
[net8_raygun_sample.zip](https://github.com/MindscapeHQ/raygun4maui/files/13540487/net8_raygun_sample.zip)
![image](https://github.com/MindscapeHQ/raygun4maui/assets/37275459/7d023014-de77-41bb-a4f5-b43d8ddab486)

issues in iOS
1. If exception is asynchronous inside of a Task in iOS, it will throw EnsureUIThread exception 
![image](https://github.com/MindscapeHQ/raygun4maui/assets/37275459/9dc7379d-5244-4e64-810c-83746d6b5190)
2. (Not in this sample project) If there is any objective-c and MAUI View Handler/Renderer exception, it will not log into raygun

Note: Android is fine

**Comment by mattleibow** [comment]
I noticed this a well when using threads other than the main/UI thread.

```cs
var thread = new Thread(() =>
{
    throw new InvalidOperationException("From a new thread.");
});
thread.Start();
```

**Comment by ProRedCat** [comment]
This should be fixed in our 1.4.2-pre-1 release for Raygun4Maui https://www.nuget.org/packages/Raygun4Maui/1.4.2-pre-1
